### PR TITLE
test: add nestscan discovery tests

### PR DESCRIPTION
**Pull request by ajcasagrande**
Signed-off-by: Anthony Casagrande <anthony.j.casagrande@intel.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [X] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- **N/A** I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: internal/netscan/util_test.go [hunk 1]
```diff
@@ -63,108 +61,178 @@ func ipGeneratorTest(input inetTest) (result inetTestResult) {
...
+			size: 0,
 		},
 		{
-			inet:      "/24",
-			expectErr: true,
+			name: "skip subnet-zero",
+			inet: mustParseCIDR(t, "192.168.1.100/0"),
+			size: 0,
+		},
+		//{
```
Comment (brianointel) [inline internal/netscan/util_test.go]: do we need these anymore?

Comment (ajcasagrande) [reply internal/netscan/util_test.go]: no, good catch.

**Review by brianointel — DISMISSED** [review]
looks good!

**Review by ChristianDarr-personal — DISMISSED** [review]
Is there a way to test cancelling the context? There are a few execution paths in the code which pertain to that,. so it might be worthwhile

**Comment by ajcasagrande** [comment]
> Is there a way to test cancelling the context? There are a few execution paths in the code which pertain to that,. so it might be worthwhile

Somewhat. I believe I added a single context cancellation test (timeout I believe), but was not sure if it was worth the effort to hit all of the other ones. Part of that is due to figuring out how to get the timing right to cancel at each point in time.

**Review by brianointel — DISMISSED** [review]
looks good

**Review by brianointel — APPROVED** [review]
looks good

**Pull request merged**